sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusScalerProvider.cs

```diff
@@ -0,0 +1,124 @@
+            ServiceBusMetadata serviceBusMetadata = JsonConvert.DeserializeObject<ServiceBusMetadata>(_triggerMetadata.Metadata.ToString());
+            AzureComponentFactory azureComponentFactory = GetAzureComponentFactory(_triggerMetadata);
+
+            (ServiceBusAdministrationClient adminClient, ServiceBusReceiver receiver) = CreateParameters(_configuration, serviceBusMetadata, azureComponentFactory);
+
+            return new ServiceBusScaleMonitor(
+                _triggerMetadata.FunctionName,
+                receiver.EntityPath,
+                !string.IsNullOrEmpty(serviceBusMetadata.QueueName) ? ServiceBusEntityType.Queue : ServiceBusEntityType.Topic,
+                new Lazy<ServiceBusReceiver>(() => receiver),
+                new Lazy<ServiceBusAdministrationClient>(() => adminClient),
+                _loggerFactory);
+        }
+
+        public ITargetScaler GetTargetScaler()
+        {
+            ServiceBusMetadata serviceBusMetadata = JsonConvert.DeserializeObject<ServiceBusMetadata>(_triggerMetadata.Metadata.ToString());
+            AzureComponentFactory azureComponentFactory = GetAzureComponentFactory(_triggerMetadata);
+
+            (ServiceBusAdministrationClient adminClient, ServiceBusReceiver receiver) = CreateParameters(_configuration, serviceBusMetadata, azureComponentFactory);
+
+            return new ServiceBusTargetScaler(
+                _triggerMetadata.FunctionName,
+                receiver.EntityPath,
+                receiver.EntityPath.Contains("//") ? ServiceBusEntityType.Topic : ServiceBusEntityType.Queue,
+                new Lazy<ServiceBusReceiver>(() => receiver),
+                new Lazy<ServiceBusAdministrationClient>(() => adminClient),
+                _options.Value,
+                serviceBusMetadata.IsSessionsEnabled,
+                !string.Equals(serviceBusMetadata.Cardinality, "many"),
+                _loggerFactory);
+        }
+
+        private (ServiceBusAdministrationClient AdminClient, ServiceBusReceiver Receiver) CreateParameters(IConfiguration configuration, ServiceBusMetadata serviceBusMetadata, AzureComponentFactory azureComponentFactory)
+        {
+            ServiceBusClientFactory factory = new ServiceBusClientFactory(configuration, azureComponentFactory, _messagingProvider, _logForwarder, _options);
+
+            ServiceBusAdministrationClient adminClient = factory.CreateAdministrationClient(serviceBusMetadata.Connection);
+            ServiceBusClient client = factory.CreateClientFromSetting(serviceBusMetadata.Connection);
+            ServiceBusReceiver receiver = !string.IsNullOrEmpty(serviceBusMetadata.QueueName) ? client.CreateReceiver(serviceBusMetadata.QueueName)
+                : client.CreateReceiver(serviceBusMetadata.TopicName, serviceBusMetadata.SubscriptionName);
+
+            return (AdminClient: adminClient, Receiver: receiver);
+        }
+
+        private AzureComponentFactory GetAzureComponentFactory(TriggerMetadata triggerMetadata)
```

Comment:
Why do we need to call this method twice, rather than once in the ctor? Both the existing ctor code and this code is pulling the factory from Properties.

Reply:
yes, the method is no needed.